File: Android/app/src/main/kotlin/com/zhechengqi/tollcat/CredentialStore.kt
package com.zhechengqi.tollcat

/**
 * 对齐 `MeterPersistence.CredentialStore`。Kotlin UI 只认协议；密钥进 Android Keystore。
 */
interface CredentialStore {
    fun save(secret: String, reference: String)
    fun read(reference: String): String?
    fun delete(reference: String)
    fun deleteAll()
}

File: Packages/MeterKit/Sources/MeterPersistence/CredentialStore.swift
import Foundation

public protocol CredentialStore: Sendable {
    func save(_ secret: String, reference: String) throws
    func read(reference: String) throws -> String?
    func delete(reference: String) throws
    /// 清除全部数据时用。按 reference 删完仍可能留下历史 orphan。
    func deleteAll() throws
}
